docs/en/sql-reference/functions/ext-dict-functions.md

```diff
@@ -481,44 +443,1954 @@ SELECT dictGetAll('regexp_dict', 'tag', 'foobarbaz', 2);
+)
+PRIMARY KEY id
+SOURCE(CLICKHOUSE(TABLE 'countries'))
+LAYOUT(FLAT())
+LIFETIME(MIN 0 MAX 1000);
+
+SELECT dictGetInt8('country_codes', 'code_designation', '3') AS code_designation_output;
+```
+
+Result:
+
+```response
+28
+```
+
+## dictGetInt8OrDefault
+
+Retrieves type Int8 values from a dictionary. If the Int8 attribute cannot be found, returns the specified default value instead.
+
+**Syntax**
+
+```sql
+dictGetInt8OrDefault('dictionary_name', 'attribute_name', key, default_value_expr)
 ```
 
 **Arguments**
 
-- `dict_name` — Name of the dictionary. [String literal](../../sql-reference/syntax.md#syntax-string-literal).
-- `attr_name` — Name of the column of the dictionary. [String literal](../../sql-reference/syntax.md#syntax-string-literal).
-- `id_expr` — Key value. [Expression](../../sql-reference/syntax.md#syntax-expressions) returning a [UInt64](../../sql-reference/data-types/int-uint.md) or [Tuple](../../sql-reference/data-types/tuple.md)-type value depending on the dictionary configuration.
-- `default_value_expr` — Value returned if the dictionary does not contain a row with the `id_expr` key. [Expression](../../sql-reference/syntax.md#syntax-expressions) returning the value in the data type configured for the `attr_name` attribute.
+- `dictionary_name`: the name of the dictionary you want to retrieve data from. [String literal](../syntax#syntax-string-literal)
+- `attribute_name`: the attribute whose value you want to get. [String literal](../syntax#syntax-string-literal), or [tuple](../data-types/tuple) of column names.
+- `key`: the key associated with the attribute. [Expression](../syntax#syntax-expressions) returning dictionary key-type value.
+- `default_value_expr` — Values returned if the dictionary does not contain a row with the `id_expr` key. [Expression](../../sql-reference/syntax.md#syntax-expressions) or [Tuple](../../sql-reference/data-types/tuple.md)([Expression](../../sql-reference/syntax.md#syntax-expressions)), returning the value (or values) in the data types configured for the `attr_names` attribute.
 
 **Returned value**
 
-- If ClickHouse parses the attribute successfully in the [attribute’s data type](../../sql-reference/dictionaries/index.md#dictionary-key-and-fields#ext_dict_structure-attributes), functions return the value of the dictionary attribute that corresponds to `id_expr`.
+The value of the dictionary attribute that corresponds to the given key. This value is parsed and returned as an `Int8` type. If the attribute cannot be found, the value of `default_value_expr` is returned instead.
 
-- If there is no requested `id_expr` in the dictionary then:
+**Implementation details**
 
-        - `dictGet[Type]` returns the content of the `<null_value>` element specified for the attribute in the dictionary configuration.
-        - `dictGet[Type]OrDefault` returns the value passed as the `default_value_expr` parameter.
+This function will throw an exception if the key does not exist in the dictionary.
```

Comment:
This is not true, it's `OrDefault` function, it will return default value in this case. But it may throw an excetpion if attribute is not `Int8` type or if default value is not `Int8` or cannot be parsed as `Int8`